beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -334,16 +334,16 @@ impl<E: EthSpec> KzgVerifiedBlobList<E> {
         kzg: &Kzg,
         seen_timestamp: Duration,
     ) -> Result<Self, KzgError> {
-        let blobs = blob_list.into_iter().collect::<Vec<_>>();
-        verify_kzg_for_blob_list(blobs.iter(), kzg)?;
+        let blobs = blob_list
+            .into_iter()
+            .map(|blob| KzgVerifiedBlob {
+                blob,
+                seen_timestamp,
+            })
+            .collect::<Vec<_>>();
+        verify_kzg_for_blob_list(blobs.iter().map(|b| &b.blob), kzg)?;
```

Comment:
This is a small optimisation to avoid allocating two `Vec`s.

Reply:
Pulled this out into a separate PR to make this PR easier to review:

- https://github.com/sigp/lighthouse/pull/6282